src/buffer_manager.ts

```diff
@@ -37,6 +40,8 @@ const LOG_PREFIX = "BufferManager";
 
 const BUFFER_NAME_PREFIX = "__vscode_neovim__-";
 
+const BUFFER_SCHEME = "vscode-neovim";
```

Comment:
Yeah, I didn't look in too much detail at how it's used, since I wanted to focus on external buffers for this PR, but I think using VScode URI scheme is the more idiomatic way to work when possible.

Right now, the neovim buffer names have a couple possible formats, as far as I can tell:
* `__vscode_neovim__-file:///path/to/file`
* `__vscode_neovim__-untitled:Untitled-1`
* `__vscode_neovim__-output:extension-output-asvetliakov.vscode-neovim-%231-vscode-neovim`
* Just `file` (e.g. `motion.txt`) for external buffers (also always unlisted, I think)

That is just the buffer name in Neovim though, and gets some translation when brought into VSCode. It seems a bit convoluted how everything is wired up, but I think if a full custom editor was implemented it might make sense to have everything under the `vscode-neovim:` scheme. For now, there is still a need for the `file://`, `output:` etc. schemes to correlate from neovim -> VSCode.

I think some bigger refactors are probably needed, and if that results in a 1-to-1 translation from neovim buffer name <-> VSCode URI it will probably simplify a lot of the buffer manager logic.